src/components/pages/block/fetch-block-details.ts

```diff
@@ -0,0 +1,52 @@
+import { l2PublicClient } from "@/lib/chains";
+import { Block, fromViemBlock, fromPrismaBlock } from "@/lib/types";
+import { prisma } from "@/lib/prisma";
+
+type FetchBlockDetailsReturnType = {
+  block: Block | null;
```

Comment:
Since these functions returns only a Block or null, there's no need for this helper type, just specify their return type as `Block | null` directly.